core-android/src/main/java/com/uber/sdk/android/core/auth/SsoDeeplink.java

```diff
@@ -106,10 +107,22 @@ public void execute() {
         intent.setData(deepLinkUri);
 
         List<PackageInfo> validatedPackages = new ArrayList<>();
-        validatedPackages.addAll(appProtocol.getInstalledPackages(activity, UBER, MIN_UBER_RIDES_VERSION_SUPPORTED));
-        validatedPackages.addAll(appProtocol.getInstalledPackages(activity, UBER_EATS, MIN_UBER_EATS_VERSION_SUPPORTED));
+        if (productFlowPriority.isEmpty()) {
+            validatedPackages.addAll(appProtocol.getInstalledPackages(activity, UBER, MIN_UBER_RIDES_VERSION_SUPPORTED));
+        } else {
+            for (SupportedAppType supportedAppType : productFlowPriority) {
+                switch (supportedAppType) {
+                    case UBER:
+                        validatedPackages.addAll(appProtocol.getInstalledPackages(activity, UBER, MIN_UBER_RIDES_VERSION_SUPPORTED));
+                        break;
+                    case UBER_EATS:
+                        validatedPackages.addAll(appProtocol.getInstalledPackages(activity, UBER_EATS, MIN_UBER_EATS_VERSION_SUPPORTED));
+                        break;
+                }
+            }
+        }
 
-        if(!validatedPackages.isEmpty()) {
+        if (!validatedPackages.isEmpty()) {
             intent.setPackage(validatedPackages.get(0).packageName);
         }
         activity.startActivityForResult(intent, requestCode);
```

Comment:
I know this was here before, do we still want this behavior? My intuition says that we should not be executing the intent if we didn't manage to validate that they have the expected app versions installed

@tyvsmith any thoughts on this?

Reply:
If there are two and we don't specify one, it'll show the intent picker which is a bad UX. If there are 0, it would have failed at the beginning of the method.

Reply:
:+1: